packages/mobile/src/dappkit/dappkit.ts

```diff
@@ -87,7 +86,7 @@ function* produceTxSignature(action: RequestTxSignatureAction) {
         params.to = tx.to
       }
       Logger.debug(TAG, 'Signing tx with params', JSON.stringify(params))
-      const signedTx = await contractKit.web3.eth.signTransaction(params)
```

Comment:
My only concern about these cases (they are all over the refactor), is that your signing directly from the wallet.

So if in the future this changes, you are going to need a refactor for this.
The idea behind using the `web3.eth.sign` is that you could switch the wallet whenever you want, without changing the code implementation.

Again, don't know if it's a problem now, but one of the advantages of using web3 is that will act as a interface, where you'll be switching the wallets, without changing anything.

Another thing, is that this asumes that the every time you want a Tx to be signed, will be from the local `geth` node . The idea of the `celoProvider` of catching the Tx that you have a local way of signing it or otherwise sent it to the node, is lost using the wallet this way. Again, is it something that could happen? A user using the remote node keystore to sign? I don't know, but this way you are removing that option.

Reply:
reverted these changes